Looks like `liburing` changed how the configure script should be called:
```
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:04.889 remove directory /root/parts/liburing/build
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:04.985 :: ++ uname -m
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:04.986 :: + '[' x86_64 '!=' x86_64 ']'
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:04.986 :: + craftctl default
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.485 Executing PosixPath('/root/parts/liburing/run/build.sh')
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.486 :: + '[' '!' -f ./configure ']'
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.487 :: + '[' '!' -f ./configure ']'
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.487 :: + '[' '!' -f ./configure ']'
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.487 :: + ./configure --prefix=
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.492 :: realpath: missing operand
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.492 :: Try 'realpath --help' for more information.
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.493 :: error:
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.493 :: Traceback (most recent call last):
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.493 ::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 111, in main
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.493 ::     ret = CraftCtl.run(cmd, args)
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.493 ::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 45, in run
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.493 ::     _client(cmd, args)
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.493 ::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 98, in _client
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.493 ::     raise RuntimeError(message)
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.493 :: RuntimeError: Failed to run the build script for part 'liburing'.
2024-09-18 00:21:06.103 :: 2024-09-18 00:21:05.694 'override-build' in part 'liburing' failed with code 1.
```

--- a/snapcraft.yaml
+++ b/snapcraft.yaml
@@ -598,8 +598,9 @@ parts:
 
   nvidia-container:
     source: https://github.com/NVIDIA/libnvidia-container
-    source-commit: d2eb0afe86f0b643e33624ee64f065dd60e952d4 # v1.14.6
+    # XXX: fails to build if using source-commit
     source-depth: 1
+    source-tag: v1.16.1
     source-type: git
     plugin: make
     build-packages:

--- a/snapcraft.yaml
+++ b/snapcraft.yaml
@@ -1411,7 +1411,7 @@ parts:
       ln -s "$(pwd)" "${GOPATH}/src/github.com/canonical/lxd"
 
       # Download the dependencies
-      go get -d -v ./...
+      go get -v ./...
     override-build: |
       set -ex
 

--- a/snapcraft.yaml
+++ b/snapcraft.yaml
@@ -738,7 +738,8 @@ parts:
       - spice-protocol
     source: https://gitlab.freedesktop.org/spice/spice
     source-commit: 0c2c1413a8b387ea597a95b6c867470a7c56c8ab # v0.15.2
-    source-depth: 1
+    # XXX: source-depth does not work with submodules
+    #source-depth: 1
     source-type: git
     plugin: meson
     meson-parameters:

